drivers/sensor/si7060/si7060.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2019 Actinius
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <i2c.h>
+#include <sensor.h>
+#include <logging/log.h>
+
+#include "si7060.h"
+
+#define SIGN_BIT_MASK 0x7F
+
+LOG_MODULE_REGISTER(si7060, CONFIG_SENSOR_LOG_LEVEL);
+
+struct si7060_data {
+	struct device *i2c_dev;
+	u16_t temperature;
+};
+
+static int si7060_reg_read(struct si7060_data *drv_data, u8_t reg,
+			     u8_t *val, bool send_stop)
+{
+	struct i2c_msg msgs[2] = {
+		{
+			.buf = &reg,
+			.len = 1,
+			.flags = I2C_MSG_WRITE,
+		},
+		{
+			.buf = val,
+			.len = 1,
+			.flags = I2C_MSG_READ,
+		},
+	};
+
+	if (send_stop) {
+		msgs[1].flags |= I2C_MSG_STOP;
+	}
+
+	if (i2c_transfer(drv_data->i2c_dev, msgs,
```

Comment:
`i2c_reg_read_byte` would be better here, as you would not have to manually construct the msgs array

Reply:
Updated.